src/libstd/process.rs

```diff
@@ -253,6 +253,10 @@ impl Command {
     /// Builder methods are provided to change these defaults and
     /// otherwise configure the process.
     ///
+    /// The `PATH` from the Command's environment will be searched
+    /// if the program contains only a single path component, as for
+    /// `execvp`.
```

Comment:
>    If program is not an absolute path, the PATH will be searched in an OS-defined way.

Something like that seems reasonable. I think the main thing that we need to include is that the behaviour is OS dependant.